protocol/shannon/sanctioned_endpoints_store.go

```diff
@@ -187,69 +187,40 @@ func (ses *sanctionedEndpointsStore) isSanctioned(endpoint *endpoint) (bool, str
 
 // --------- Session Sanction Key ---------
 
-// TODO_TECHDEBT(@commoddity,adshmh): update this to be composed of only
-//     - endpoint URL
-//     AND
-//     - either supplier address or session ID
-// Discord conversation: https://discord.com/channels/824324475256438814/1273320783547990160/1378346761151844465
-
-// sanctionKey:
-//   - Creates a unique key for session-based sanctions
-//   - Format: "<app_address>:<session_id>:<supplier_address>:<endpoint_url>"
-type sanctionKey struct {
-	supplier    string
-	endpointURL string
-	appAddr     string
-	sessionID   string
-}
-
-func newSanctionKey(endpoint *endpoint) sanctionKey {
-	// Session header is never nil:
-	// Ref: https://github.com/pokt-network/shannon-sdk/blob/64d83f85e7e3f8e7d6ddee98ced276203cf5475f/session.go#L134
-	header := endpoint.session.Header
-
-	appAddr := header.ApplicationAddress
-	sessionID := header.SessionId
-	return sanctionKey{
-		supplier:    endpoint.supplier,
-		endpointURL: endpoint.url,
-		appAddr:     appAddr,
-		sessionID:   sessionID,
-	}
+// newSessionSanctionKey creates a key for a session-based sanction.
+// A sanction key is a string composed of the endpoint address and the session ID, separated by a hyphen.
+// The session ID is appended to ensure that session sanctions do not extend beyond the session duration.
+//
+// Example for the key "pokt1ggdpwj5stslx2e567qcm50wyntlym5c4n0dst8-https://im.oldgreg.org-1234567890":
+//   - Endpoint address (supplier address + endpoint URL): "pokt1ggdpwj5stslx2e567qcm50wyntlym5c4n0dst8-https://im.oldgreg.org"
+//   - Session ID: "1234567890"
+//
+// The key is used to store and retrieve session-based sanctions from the cache.
+func newSessionSanctionKey(endpoint *endpoint) string {
+	endpointAddr := endpoint.Addr()
+	sessionID := endpoint.session.Header.SessionId
+	return fmt.Sprintf("%s-%s", endpointAddr, sessionID)
 }
 
-// sanctionKeyFromCacheKeyString decomposes the cache key into its components
-// in order to populate the details map with the data required.
-func sanctionKeyFromCacheKeyString(cacheKey string) sanctionKey {
-	// Only split for 4 parts, as final part is URL which contains a ":" character.
-	parts := strings.SplitN(cacheKey, ":", 4)
-	fmt.Println("🚀parts", parts)
-	fmt.Println("cacheKey", cacheKey)
-	return sanctionKey{
-		appAddr:     parts[0],
-		sessionID:   parts[1],
-		supplier:    parts[2],
-		endpointURL: parts[3],
+// decomposeSessionSanctionKey decomposes a sanction key into its components.
+// It returns the endpoint address and the session ID.
+//
+// Example:
+// - Full Key: pokt1ggdpwj5stslx2e567qcm50wyntlym5c4n0dst8-https://im.oldgreg.org-1234567890
+//   - Endpoint address: "pokt1ggdpwj5stslx2e567qcm50wyntlym5c4n0dst8-https://im.oldgreg.org"
+//   - Session ID: "1234567890"
+func decomposeSessionSanctionKey(key string) (protocol.EndpointAddr, string) {
```

Comment:
This does not look GO idiomatic: How about using a sessionSanctionKey struct with a String() method instead?